src/backend/distributed/master/master_modify_multiple_shards.c

```diff
@@ -290,21 +366,25 @@ SendQueryToPlacements(char *shardQueryString, ShardConnections *shardConnections
 		}
 
 		placementAffectedTupleString = PQcmdTuples(result);
-		placementAffectedTupleCount = pg_atoi(placementAffectedTupleString,
-											  sizeof(int32), 0);
 
-		if ((shardAffectedTupleCount == -1) ||
-			(shardAffectedTupleCount == placementAffectedTupleCount))
+		if (returnTupleCount)
```

Comment:
right, did very similar
used
``
placementAffectedTupleCount = 0;
``
instead of
``
placementAffectedTupleCount = unknownTupleCount;
``